shm/item.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include "cargo.hpp"
+#include "rarity.hpp"
+#include "timeSHM.hpp"
+
+class Item : public Cargo, public Observer {
+public:
+    Item(std::string name, size_t amount, size_t basePrice, int rarity, Time* Publisher);
+
+    ~Item() { this->Publisher_->removeObserver(this); };
```

Comment:
That's why you should move all definition to .cpp. (see my comment about unregisterObserver)